### DISCUSSION TITLE
Proposal to Rename Fishermen

### DISCUSSION THREAD

**Discussion by dylanlott** [pokt-network/pocket#701] [General]
[ADR 0005 - Rename Fishermen](https://github.com/pokt-network/pocket-network-protocol/pull/38) proposes renaming the entity in the POKT Network Protocol "fishermen". However, the ADR does not yet have a decision listed.

This discussion is to open up the floor to debate for and against renaming Fishermen in the POKT Network protocol and code as a whole.

<img src="https://media4.giphy.com/media/UWJ1aUoireVVDKayR5/giphy.gif"/>

To start, the current proposed names according to ADR 0005 are:

- Auditor
- Checker
- Verifier
- Sampler
- Watcher

**Comment by Olshansk** [comment]
I wouldn't actually call this controversial, because I agree! Figuring out where/when to use `fishermen` vs `fisherman` vs `fisherperson` vs `fisherbeing` can get inconsistent very easily.

For historical context, we originally adopted the term `Fisherman` per Polkadot's naming convention [here](https://wiki.polkadot.network/docs/learn-availability).

<img width="987" alt="Screenshot 2023-04-26 at 4 14 45 PM" src="https://user-images.githubusercontent.com/1892194/234722229-f6c5881f-2ef1-4c40-9427-696c16ae0b63.png">

My personal vote is for `Watcher` for two reasons:
1. It's intuitive, makes sense, easy to remember, etc...
2. We can use gifs like this when a Watcher unstakes:

![giphy](https://user-images.githubusercontent.com/1892194/234722578-b29228c6-569e-457d-af90-b61af764ebfc.gif)

**Reply by bryanchriswhite** [reply]
:rofl:

**Reply by okdas** [reply]
How about "BigBrother"? 😅

<img src="https://media1.giphy.com/media/attnKlsj4LH6k6df3v/giphy.gif"/>

**Comment by ghost** [comment]
The Verifier �
![DetectivePikachuInvestigationGIF](https://user-images.githubusercontent.com/122069876/235260582-67950e10-8a44-46ea-a60a-d2ef5923f16d.gif)
�️‍♀️

**Comment by alleneubank** [comment]
I like Watcher, but would like to throw another name in the ring, Enforcer.

**Reply by Olshansk** [reply]
Can't wait for when Pocket is going to power WWE

**Comment by h5law** [comment]
I think "Watcher" is meant to be, check out #️⃣5️⃣⬇️  👀 

![image](https://user-images.githubusercontent.com/53987565/235450871-1da39d16-3fd4-4997-8ddc-826de58c8e0c.png)

**Reply by ghost** [reply]
#5 for h5?

**Reply by jessicadaugherty** [reply]
Ranger is really cool though ngl

**Comment by dylanlott** [comment]
Watcher has emerged as a current favorite. I'm going to move ahead with updating this document to reflect the decision and next steps to be taken for systematic replacement in the rest of the codebase. @Olshansk any thoughts here?

**Reply by Olshansk** [reply]
Yes, let's do it! In terms of steps, let's:

1. Update and merge in the ADR https://github.com/pokt-network/pocket-network-protocol/pull/38/files
2. Update the specification https://github.com/pokt-network/pocket-network-protocol/
3. Update the codebase https://github.com/pokt-network/pocket

https://www.goodreads.com/quotes/295047-night-gathers-and-now-my-watch-begins-it-shall-not

![Screenshot 2023-05-16 at 3 33 04 PM](https://github.com/pokt-network/pocket/assets/1892194/f0ab5219-a1af-4bc2-bf31-8c62e93f2bf4)

**Comment by Olshansk** [comment]
@dylanlott Found another reference to the term `Fisherman` as someone who raises the alarm in a 2018 article by Ethereum around the topic of fraud proofs and data availability. There's a whole problem called `The Fisherman's Dilemma

Doesn't change anything but just sharing for fun.

Reference: https://github.com/ethereum/research/wiki/A-note-on-data-availability-and-erasure-coding

> Here, fraud proofs are not a solution, because not publishing data is not a uniquely attributable fault - in any scheme where a node ("fisherman") has the ability to "raise the alarm" about some piece of data not being available, if the publisher then publishes the remaining data, all nodes who were not paying attention to that specific piece of data at that exact time cannot determine whether it was the publisher that was maliciously withholding data or whether it was the fisherman that was maliciously making a false alarm.

![fisherman_dilemma (1)](https://github.com/pokt-network/pocket/assets/1892194/77114b4d-de6d-4be7-b2af-54bb2835e561)
![fisherman_dilemma_1 (1)](https://github.com/pokt-network/pocket/assets/1892194/2f08db1a-f8d6-46bc-a34c-3b9d4aeceb03)

**Comment by Olshansk** [comment]
@0xRampey You down to take this one on and rename "all the things"?
![all-the-things-hyperbole-and-a-half](https://github.com/pokt-network/pocket/assets/1892194/3a182622-0639-4e75-bdc1-3435c5a26d27)

**Reply by 0xRampey** [reply]
<img src="https://media1.giphy.com/media/Lm63QU87HvgVEuTV63/giphy.gif"/>

**Reply by Olshansk** [reply]
https://github.com/pokt-network/pocket/issues/956

**Comment by 0xRampey** [comment]
The roadmap for implementing Watchers still has `Fishermen`-related names for each of the milestones.

<img width="1076" alt="Screenshot 2023-08-09 at 3 14 40 PM" src="https://github.com/pokt-network/pocket/assets/2914233/7aad2059-b877-46ab-b193-8863f308a1c1">

Following up on the above discussion, the floor is yet again open for milestones inspired by the Night's watch. 
Here are some AI-assisted suggestions that sound great to me.

- Proto-fish -> First Raven
- Castaway -> Crow's council
- Fishermen -> Night's Watch
- Cast-net -> Wall's Vigil
